src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,163 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private MyNode head;
+    private MyNode tail;
+    private int size;
+
+    public MyLinkedList() {
+        head = null;
+        tail = null;
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
+        MyNode myNode = new MyNode(value);
+        if (size == 0) {
+            head = myNode;
+            tail = myNode;
+        } else {
+            tail.setNext(myNode);
+            myNode.setPrev(tail);
+            tail = myNode;
```

Comment:
Avoid using getters and setters for `MyNode` as per the checklist: 'Don't use getters and setters in class Node.'